### ISSUE TITLE
NativeBlockOutputStream: invalid cast from ColumnNullable to ColumnVector

### ISSUE THREAD

**Issue by tavplubix** [ClickHouse/ClickHouse#21002]
https://clickhouse-test-reports.s3.yandex.net/20946/f5fc082ed0e4b3d9f412f6ecfb5db0ce49469d3e/fuzzer_ubsan/report.html#fail1

```
2021.02.20 04:37:46.668035 [ 216 ] {5535e0e0-de60-48d4-96bf-26ed7b04ff87} <Debug> executeQuery: (from [::1]:36844, using production parser) SELECT EventDate FROM (SELECT EventDate FROM tracking_events_tmp AS t1 WHERE (EventDate >= toDate('2020-07-10')) AND (EventDate <= toDate('2020-07-11')) AND (APIKey = 2) GROUP BY APIKey = 2, EventDate) FULL OUTER JOIN (SELECT EventDate FROM remote('127.0.0.{1,3}', currentDatabase(), open_events_tmp) AS t2 WHERE (EventDate >= toDate('2020-07-10')) AND (EventDate <= toDate('2020-07-11')) AND (APIKey = 2) GROUP BY EventDate WITH TOTALS) USING (EventDate) ORDER BY EventDate ASC NULLS FIRST SETTINGS totals_mode = 'after_having_auto', group_by_overflow_mode = 'any', max_rows_to_group_by = 10000000, joined_subquery_requires_alias = 0

...

../src/DataTypes/DataTypeNullable.cpp:93:9: runtime error: member call on address 0x7fe4e000d190 which does not point to an object of type 'DB::ColumnNullable'
0x7fe4e000d190: note: object is of type 'DB::ColumnVector<unsigned short>'
 00 00 00 00  f0 39 ae 08 00 00 00 00  03 00 00 00 e4 7f 00 00  40 dd 00 e0 e4 7f 00 00  42 dd 00 e0
              ^~~~~~~~~~~~~~~~~~~~~~~
              vptr for 'DB::ColumnVector<unsigned short>'
    #0 0x18f00c9f in DB::DataTypeNullable::serializeBinaryBulkWithMultipleStreamsImpl(DB::IColumn const&, unsigned long, unsigned long, DB::IDataType::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::IDataType::SerializeBinaryBulkState>&) const obj-x86_64-linux-gnu/../src/DataTypes/DataTypeNullable.cpp:93:9
    #1 0x197309f3 in DB::NativeBlockOutputStream::writeData(DB::IDataType const&, COW<DB::IColumn>::immutable_ptr<DB::IColumn> const&, DB::WriteBuffer&, unsigned long, unsigned long) obj-x86_64-linux-gnu/../src/DataStreams/NativeBlockOutputStream.cpp:58:10
    #2 0x197314ef in DB::NativeBlockOutputStream::write(DB::Block const&) obj-x86_64-linux-gnu/../src/DataStreams/NativeBlockOutputStream.cpp:124:13
    #3 0x1a415494 in DB::TCPHandler::sendTotals(DB::Block const&) obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:763:26
    #4 0x1a40f664 in DB::TCPHandler::processOrdinaryQueryWithProcessors() obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:678:13
    #5 0x1a403072 in DB::TCPHandler::runImpl() obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:306:17
    #6 0x1a41c52c in DB::TCPHandler::run() obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1492:9
    #7 0x1b327c2b in Poco::Net::TCPServerConnection::start() obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:43:3
    #8 0x1b328082 in Poco::Net::TCPServerDispatcher::run() obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerDispatcher.cpp:113:19
    #9 0x1b4057f9 in Poco::PooledThread::run() obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/ThreadPool.cpp:199:14
    #10 0x1b4035fb in Poco::ThreadImpl::runnableEntry(void*) obj-x86_64-linux-gnu/../contrib/poco/Foundation/src/Thread_POSIX.cpp:345:27
    #11 0x7fe7b9524608 in start_thread /build/glibc-eX1tMB/glibc-2.31/nptl/pthread_create.c:477:8
    #12 0x7fe7b944b292 in clone /build/glibc-eX1tMB/glibc-2.31/misc/../sysdeps/unix/sysv/linux/x86_64/clone.S:95

SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/DataTypes/DataTypeNullable.cpp:93:9 in 
2021.02.20 04:37:46.698580 [ 87 ] {} <Trace> BaseDaemon: Received signal -3
2021.02.20 04:37:46.698831 [ 218 ] {} <Fatal> BaseDaemon: ########################################
2021.02.20 04:37:46.698893 [ 218 ] {} <Fatal> BaseDaemon: (version 21.3.1.6067, build id: B050D13BB7640A02C30589ECF3C24DF6DA1BF39A) (from thread 216) (query_id: 5535e0e0-de60-48d4-96bf-26ed7b04ff87) Received signal Unknown signal (-3)
2021.02.20 04:37:46.698934 [ 218 ] {} <Fatal> BaseDaemon: Sanitizer trap.
2021.02.20 04:37:46.698995 [ 218 ] {} <Fatal> BaseDaemon: Stack trace: 0xdb2ebd5 0xdd5abe6 0xdaf2a96 0xdb03579 0x18f00ca0 0x197309f4 0x197314f0 0x1a415495 0x1a40f665 0x1a403073 0x1a41c52d 0x1b327c2c 0x1b328083 0x1b4057fa 0x1b4035fc 0x7fe7b9524609 0x7fe7b944b293
2021.02.20 04:37:46.719175 [ 218 ] {} <Fatal> BaseDaemon: 0.1. inlined from ./obj-x86_64-linux-gnu/../src/Common/StackTrace.cpp:298: StackTrace::tryCapture()
2021.02.20 04:37:46.719204 [ 218 ] {} <Fatal> BaseDaemon: 0. ../src/Common/StackTrace.cpp:259: StackTrace::StackTrace() @ 0xdb2ebd5 in /workspace/clickhouse
2021.02.20 04:37:46.781799 [ 218 ] {} <Fatal> BaseDaemon: 1. ./obj-x86_64-linux-gnu/../base/daemon/BaseDaemon.cpp:381: sanitizerDeathCallback() @ 0xdd5abe6 in /workspace/clickhouse
2021.02.20 04:37:47.274616 [ 135 ] {} <Trace> SystemLog (system.metric_log): Flushing system log, 8 entries to flush
2021.02.20 04:37:47.285021 [ 135 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 509.01 GiB.
2021.02.20 04:37:47.289979 [ 135 ] {} <Trace> system.metric_log (8a6f9b0e-21d9-40f7-b2fd-0c4a037dd881): Renaming temporary part tmp_insert_202102_162_162_0 to 202102_162_162_0.
2021.02.20 04:37:47.290894 [ 135 ] {} <Trace> SystemLog (system.metric_log): Flushed system log
2021.02.20 04:37:47.683124 [ 218 ] {} <Fatal> BaseDaemon: 2. __sanitizer::Die() @ 0xdaf2a96 in /workspace/clickhouse
2021.02.20 04:37:47.957663 [ 203 ] {} <Trace> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Found 8 old parts to remove.
2021.02.20 04:37:47.957739 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_53_68_2
2021.02.20 04:37:47.960892 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_69_90_3
2021.02.20 04:37:47.964160 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_91_91_0
2021.02.20 04:37:47.964465 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_92_92_0
2021.02.20 04:37:47.964739 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_93_93_0
2021.02.20 04:37:47.964995 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_94_94_0
2021.02.20 04:37:47.965254 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_95_95_0
2021.02.20 04:37:47.965509 [ 203 ] {} <Debug> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Removing part from filesystem 202102_96_96_0
2021.02.20 04:37:48.276599 [ 134 ] {} <Trace> SystemLog (system.query_thread_log): Flushing system log, 23 entries to flush
2021.02.20 04:37:48.278481 [ 134 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 509.03 GiB.
2021.02.20 04:37:48.279971 [ 134 ] {} <Trace> system.query_thread_log (83dcb494-8c7f-48ce-a85d-7b307b9a52cd): Renaming temporary part tmp_insert_202102_160_160_0 to 202102_160_160_0.
2021.02.20 04:37:48.280199 [ 134 ] {} <Trace> SystemLog (system.query_thread_log): Flushed system log
2021.02.20 04:37:48.592099 [ 218 ] {} <Fatal> BaseDaemon: 3. ? @ 0xdb03579 in /workspace/clickhouse
2021.02.20 04:37:48.609092 [ 218 ] {} <Fatal> BaseDaemon: 4. ./obj-x86_64-linux-gnu/../src/DataTypes/DataTypeNullable.cpp:0: DB::DataTypeNullable::serializeBinaryBulkWithMultipleStreamsImpl(DB::IColumn const&, unsigned long, unsigned long, DB::IDataType::SerializeBinaryBulkSettings&, std::__1::shared_ptr<DB::IDataType::SerializeBinaryBulkState>&) const @ 0x18f00ca0 in /workspace/clickhouse
2021.02.20 04:37:48.613774 [ 132 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 509.03 GiB.
2021.02.20 04:37:48.615166 [ 132 ] {} <Trace> system.text_log (161ec0aa-2fa1-4e7e-9412-960fd879b1fc): Renaming temporary part tmp_insert_202102_162_162_0 to 202102_162_162_0.
2021.02.20 04:37:48.622483 [ 218 ] {} <Fatal> BaseDaemon: 5. ./obj-x86_64-linux-gnu/../src/DataStreams/NativeBlockOutputStream.cpp:59: DB::NativeBlockOutputStream::writeData(DB::IDataType const&, COW<DB::IColumn>::immutable_ptr<DB::IColumn> const&, DB::WriteBuffer&, unsigned long, unsigned long) @ 0x197309f4 in /workspace/clickhouse
2021.02.20 04:37:48.636750 [ 218 ] {} <Fatal> BaseDaemon: 6. ./obj-x86_64-linux-gnu/../src/DataStreams/NativeBlockOutputStream.cpp:126: DB::NativeBlockOutputStream::write(DB::Block const&) @ 0x197314f0 in /workspace/clickhouse
2021.02.20 04:37:48.761528 [ 218 ] {} <Fatal> BaseDaemon: 7. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:764: DB::TCPHandler::sendTotals(DB::Block const&) @ 0x1a415495 in /workspace/clickhouse
2021.02.20 04:37:48.884554 [ 218 ] {} <Fatal> BaseDaemon: 8.1. inlined from ./obj-x86_64-linux-gnu/../contrib/libcxx/include/unordered_map:1044: ~unordered_map
2021.02.20 04:37:48.884614 [ 218 ] {} <Fatal> BaseDaemon: 8.2. inlined from ../src/Core/Block.h:27: ~Block
2021.02.20 04:37:48.884660 [ 218 ] {} <Fatal> BaseDaemon: 8. ../src/Server/TCPHandler.cpp:678: DB::TCPHandler::processOrdinaryQueryWithProcessors() @ 0x1a40f665 in /workspace/clickhouse
2021.02.20 04:37:48.996795 [ 218 ] {} <Fatal> BaseDaemon: 9. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:306: DB::TCPHandler::runImpl() @ 0x1a403073 in /workspace/clickhouse
2021.02.20 04:37:49.028630 [ 133 ] {} <Trace> SystemLog (system.query_log): Flushing system log, 10 entries to flush
2021.02.20 04:37:49.031564 [ 133 ] {} <Debug> DiskLocal: Reserving 1.00 MiB on disk `default`, having unreserved 509.03 GiB.
2021.02.20 04:37:49.034005 [ 133 ] {} <Trace> system.query_log (43159764-f3f4-448a-8615-ff6d2a4d45e1): Renaming temporary part tmp_insert_202102_161_161_0 to 202102_161_161_0.
2021.02.20 04:37:49.034320 [ 133 ] {} <Trace> SystemLog (system.query_log): Flushed system log
2021.02.20 04:37:49.124978 [ 218 ] {} <Fatal> BaseDaemon: 10. ./obj-x86_64-linux-gnu/../src/Server/TCPHandler.cpp:1494: DB::TCPHandler::run() @ 0x1a41c52d in /workspace/clickhouse
2021.02.20 04:37:49.129144 [ 218 ] {} <Fatal> BaseDaemon: 11. ./obj-x86_64-linux-gnu/../contrib/poco/Net/src/TCPServerConnection.cpp:57: Poco::Net::TCPServerConnection::start() @ 0x1b327c2c in /workspace/clickhouse
2021.02.20 04:37:49.141685 [ 218 ] {} <Fatal> BaseDaemon: 12.1. inlined from ./obj-x86_64-linux-gnu/../contrib/libcxx/include/atomic:1072: int std::__1::__cxx_atomic_fetch_sub<int>(std::__1::__cxx_atomic_base_impl<int>*, int, std::__1::memory_order)
2021.02.20 04:37:49.141771 [ 218 ] {} <Fatal> BaseDaemon: 12.2. inlined from ../contrib/libcxx/include/atomic:1725: std::__1::__atomic_base<int, true>::fetch_sub(int, std::__1::memory_order)
2021.02.20 04:37:49.141862 [ 218 ] {} <Fatal> BaseDaemon: 12.3. inlined from ../contrib/libcxx/include/atomic:1760: std::__1::__atomic_base<int, true>::operator--()
2021.02.20 04:37:49.141913 [ 218 ] {} <Fatal> BaseDaemon: 12.4. inlined from ../contrib/poco/Net/src/TCPServerDispatcher.cpp:231: Poco::Net::TCPServerDispatcher::endConnection()
2021.02.20 04:37:49.141965 [ 218 ] {} <Fatal> BaseDaemon: 12. ../contrib/poco/Net/src/TCPServerDispatcher.cpp:114: Poco::Net::TCPServerDispatcher::run() @ 0x1b328083 in /workspace/clickhouse
2021.02.20 04:37:49.152798 [ 218 ] {} <Fatal> BaseDaemon: 13.1. inlined from ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/ScopedLock.h:36: ScopedLock
2021.02.20 04:37:49.152870 [ 218 ] {} <Fatal> BaseDaemon: 13. ../contrib/poco/Foundation/src/ThreadPool.cpp:213: Poco::PooledThread::run() @ 0x1b4057fa in /workspace/clickhouse
2021.02.20 04:37:49.162619 [ 218 ] {} <Fatal> BaseDaemon: 14.1. inlined from ./obj-x86_64-linux-gnu/../contrib/poco/Foundation/include/Poco/SharedPtr.h:156: Poco::SharedPtr<Poco::Runnable, Poco::ReferenceCounter, Poco::ReleasePolicy<Poco::Runnable> >::assign(Poco::Runnable*)
2021.02.20 04:37:49.162688 [ 218 ] {} <Fatal> BaseDaemon: 14.2. inlined from ../contrib/poco/Foundation/include/Poco/SharedPtr.h:208: Poco::SharedPtr<Poco::Runnable, Poco::ReferenceCounter, Poco::ReleasePolicy<Poco::Runnable> >::operator=(Poco::Runnable*)
2021.02.20 04:37:49.162735 [ 218 ] {} <Fatal> BaseDaemon: 14. ../contrib/poco/Foundation/src/Thread_POSIX.cpp:360: Poco::ThreadImpl::runnableEntry(void*) @ 0x1b4035fc in /workspace/clickhouse
2021.02.20 04:37:49.162814 [ 218 ] {} <Fatal> BaseDaemon: 15. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
2021.02.20 04:37:49.162887 [ 218 ] {} <Fatal> BaseDaemon: 16. clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
2021.02.20 04:37:49.425965 [ 218 ] {} <Fatal> BaseDaemon: Calculated checksum of the binary: D3CAEF4431D68E0D9F448FAD5C35D24F. There is no information about the reference checksum.
2021.02.20 04:37:51.269874 [ 82 ] {} <Information> Application: Child process exited normally with code 1.
```

**Comment by alexey-milovidov** [comment]
Most likely fixed by #26639.

**Comment by alexey-milovidov** [comment]
Or it is duplicate of something related to FULL JOIN:

```
SELECT EventDate
FROM
(
    SELECT EventDate
    FROM tracking_events_tmp AS t1
    WHERE (EventDate >= toDate('2020-07-10')) AND (EventDate <= toDate('2020-07-11')) AND (APIKey = 2)
    GROUP BY
        APIKey = 2,
        EventDate
)
FULL OUTER JOIN
(
    SELECT EventDate
    FROM remote('127.0.0.{1,3}', currentDatabase(), open_events_tmp) AS t2
    WHERE (EventDate >= toDate('2020-07-10')) AND (EventDate <= toDate('2020-07-11')) AND (APIKey = 2)
    GROUP BY EventDate
        WITH TOTALS
) USING (EventDate)
ORDER BY EventDate ASC NULLS FIRST
SETTINGS totals_mode = 'after_having_auto', group_by_overflow_mode = 'any', max_rows_to_group_by = 10000000, joined_subquery_requires_alias = 0
```
